app.js

```diff
@@ -1,18 +1,77 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
 
 // require spotify-web-api-node package here:
+const SpotifyWebApi = require("spotify-web-api-node");
 
 const app = express();
-
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
 
 // Our routes go here:
 
-app.listen(3000, () => console.log('My Spotify project running on port 3000 🎧 🥁 🎸 🔊'));
+app.get("/", (req, res) => {
+  res.render("index");
+});
+
+// Query
+
+app.get("/artist-search", (req, res) => {
+  spotifyApi
+    .searchArtists(req.query.search)
+    .then((data) => {
+      console.log(
+        "The received data from the API: ",
+        data.body.artists.items[0]
+      );
+      res.render("artist-search-results", { result: data.body.artists.items });
+    })
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+});
+
+app.get("/albums/:artistId", (req, res, next) => {
+  let {artistId} = req.params;
+  spotifyApi
+    .getArtistAlbums(artistId)
+    .then((data) => {
+      console.log("The received data from the API: ", data.body.items[0]);
+      res.render("albums", { result: data.body.items });
+    })
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+});
+
+app.get("/albums/:trackId", (req, res) => {
```

Comment:
About the last iteration, if you try to access the tracks of an album you're getting `Cannot GET /tracks/6x5UJEgPpRGdXjpeRY5QSz` 

This is happening because you're sending the user to the path `/tracks/...` but in the `app.js` file you forget to change the path of the route `app.get("/albums/:trackId", (req, res) => {` 

If you change `/albums/:trackId` to `/tracks/:trackId` it will work.